gateway/src/main/java/io/zeebe/gateway/EndpointManager.java

```diff
@@ -140,103 +109,163 @@ private void addPartitionInfoToBrokerInfo(
   }
 
   @Override
-  public void deployWorkflow(
-      final DeployWorkflowRequest request,
-      final StreamObserver<DeployWorkflowResponse> responseObserver) {
-
-    sendRequest(
-        request,
-        RequestMapper::toDeployWorkflowRequest,
-        ResponseMapper::toDeployWorkflowResponse,
-        responseObserver);
+  public void activateJobs(
+      final ActivateJobsRequest request,
+      final StreamObserver<ActivateJobsResponse> responseObserver) {
+    final BrokerClusterState topology = topologyManager.getTopology();
+    activateJobsHandler.activateJobs(topology.getPartitionsCount(), request, responseObserver);
   }
 
   @Override
-  public void publishMessage(
-      PublishMessageRequest request, StreamObserver<PublishMessageResponse> responseObserver) {
-
+  public void cancelWorkflowInstance(
+      final CancelWorkflowInstanceRequest request,
+      final StreamObserver<CancelWorkflowInstanceResponse> responseObserver) {
     sendRequest(
         request,
-        RequestMapper::toPublishMessageRequest,
-        ResponseMapper::toPublishMessageResponse,
+        RequestMapper::toCancelWorkflowInstanceRequest,
+        ResponseMapper::toCancelWorkflowInstanceResponse,
         responseObserver);
   }
 
   @Override
-  public void updateJobRetries(
-      UpdateJobRetriesRequest request, StreamObserver<UpdateJobRetriesResponse> responseObserver) {
+  public void completeJob(
+      final CompleteJobRequest request,
+      final StreamObserver<CompleteJobResponse> responseObserver) {
     sendRequest(
         request,
-        RequestMapper::toUpdateJobRetriesRequest,
-        ResponseMapper::toUpdateJobRetriesResponse,
+        RequestMapper::toCompleteJobRequest,
+        ResponseMapper::toCompleteJobResponse,
         responseObserver);
   }
 
   @Override
   public void createWorkflowInstance(
-      CreateWorkflowInstanceRequest request,
-      StreamObserver<CreateWorkflowInstanceResponse> responseObserver) {
+      final CreateWorkflowInstanceRequest request,
+      final StreamObserver<CreateWorkflowInstanceResponse> responseObserver) {
+    final BrokerCreateWorkflowInstanceRequest brokerRequest;
```

Comment:
With `ActivateJobsHandler`, I think we defer the complete logic there, and the endpoint here is quite small - should we do the same here, and let the handler also perform the mapping and error handling logic? I would lean towards yes to keep this class here more about dispatching to the right handler. Let me know what you think

Reply:
We will then have to duplicate all code for mapping errors to grpc status.